src/cfml/system/config/server.schema.json

```diff
@@ -389,6 +535,166 @@
+					"title": "Web Rules",
+					"description": "Ad-hoc rules using Undertow predicates and handlers",
+					"type": "array",
+					"items": {
+						"title": "Web Rule",
+						"description": "Ad-hoc rule using Undertow predicates and handlers",
+						"type": "string"
+					},
+					"default": []
+				},
+				"rulesFile": {
+					"title": "Web Rules File",
+					"description": "A path or paths to files containing Undertow predicates and handlers",
+					"type": [
+						"string",
+						"array"
+					],
+					"items": {
+						"title": "Web Rules File",
+						"description": "A path to file containing Undertow predicates and handlers",
+						"type": "string"
+					},
+					"default": []
+				},
+				"allowedExt":  {
+					"title": "Web Allowed Ext",
+					"description": "A comma-delimited list of additional file extensions allowed by web server",
+					"type": "string",
+					"default": ""
+				},
+				"useProxyForwardedIP":  {
+					"title": "Use Proxy Forwarded IP",
+					"description": "Whether the remote IP in your CF engine's cgi scope represents the upstream IP",
+					"type": "boolean",
+					"default": false
+				},
+				"security": {
+					"title": "Web Security",
+					"description": "Configure web security",
+					"type": "object",
+					"properties": {
+						"realm":  {
+							"title": "Realm",
+							"description": "",
+							"type": "string",
+							"default": ""
+						},
+						"authPredicate":  {
+							"title": "Auth Predicate",
+							"description": "",
+							"type": "string",
+							"default": ""
+						},
+						"basicAuth": {
+							"title": "Basic Authentication",
+							"description": "Configure basic authentication",
+							"type": "object",
+							"properties": {
+								"enable": {
+									"title": "Basic Auth Enable",
+									"description": "Enable basic auth for this server",
+									"type": "boolean"
+								},
+								"users": {
+									"title": "Basic Auth Users",
+									"description": "Users who can authenticate to basic auth",
+									"type": "object",
+									"additionalProperties": {
+										"title": "User",
+										"description": "The key is the user name and the value is the password.",
+										"type": "string"
+									}
+								}
+							}
+						},
+						"clientCert": {
+							"title": "Web Security Client Cert",
+							"description": "",
+							"type": "object",
+							"properties": {
+								"enable": {
+									"title": "Client Cert Enable",
+									"description": "",
+									"type": "boolean",
+									"default": false
+								},
+								"SSLRenegotiationEnable": {
+									"title": "Client Cert SSL Renegotiation Enable",
+									"description": "",
+									"type": "boolean",
+									"default": false
+								},
+								"trustUpstreamHeaders": {
+									"title": "Client Cert Trust Upstream Headers",
+									"description": "",
+									"type": "boolean",
+									"default": false
+								},
+								"subjectDNs": {
```

Comment:
subjectDNs and issuerDNs can be either an array or a string

Reply:
Actually looking through the code I see that now. FYI, I couldn't find any documentation on `web.security.clientCert`.

Reply:
@KamasamaK The client cert stuff is brand new and not released yet.  It is documented here:
https://ortussolutions.atlassian.net/browse/COMMANDBOX-1497
and I'll add it to the Gitbook docs on release.